src/fcc_bridge_node_safety.cpp

```diff
@@ -470,106 +478,81 @@ bool FCCBridgeNode::check_flight_mode() {
 void FCCBridgeNode::check_battery_state() {
     RCLCPP_DEBUG(this->get_safety_logger(), "Checking battery state");
 
-    switch (this->get_internal_state()) {
-            // This function should never be called in these states
-        case INTERNAL_STATE::ERROR:
-            // This should never happen, as the process exits on ERROR state
-            throw std::runtime_error(std::string(__func__) +
-                                     " called while in ERROR state");
-        case INTERNAL_STATE::STARTING_UP:
-        case INTERNAL_STATE::ROS_SET_UP:
-            RCLCPP_FATAL(this->get_internal_state_logger(),
-                         "In an invalid state for a batter check! Exiting...");
+    if (this->get_internal_state() == INTERNAL_STATE::STARTING_UP ||
+        this->get_internal_state() == INTERNAL_STATE::ROS_SET_UP) {
+        RCLCPP_FATAL(this->get_internal_state_logger(),
+                     "In an invalid state for a battery check! Exiting...");
+        this->set_internal_state(INTERNAL_STATE::ERROR);
+        this->exit_process_on_error();
+    }
+
+    if (this->get_internal_state() == INTERNAL_STATE::MAVSDK_SET_UP) {
+        // Acceptable. In this state no safety limits are configured so a
+        // check is not possible
+        return;
+    }
+
+    if (!this->safety_limits.has_value()) {
+        if (this->is_airborne()) {
+            RCLCPP_FATAL(this->get_safety_logger(),
+                         "Found no safety limits! UAV is airborne. "
+                         "Triggering RTH...");
+            this->trigger_rth();
+            return;
+        } else {
+            RCLCPP_FATAL(
+                this->get_safety_logger(),
+                "Found no safety limits! UAV is not airborne. Exiting...");
             this->set_internal_state(INTERNAL_STATE::ERROR);
             this->exit_process_on_error();
-        case INTERNAL_STATE::MAVSDK_SET_UP:
-            // Acceptable. In this state no safety limits are configured so a
-            // check is not possible
+        }
+    }
+
+    // Ensuring there is a valid BatterState present
```

Comment:
typo, "BatterState" should be "BatteryState"

Reply:
Fixed in 39a4337